### PR TITLE
Circuit Refactoring

### PR DESCRIPTION
**Pull request by maxcapodi78**
added connect_components_in_parallel and connect_components_in_series methods

### CODE REVIEW

File: pyaedt/modeler/PrimitivesNexxim.py [hunk 1]
```diff
@@ -59,12 +59,88 @@ def __init__(self, modeler):
...
+        for component in components_to_connect:
+            if isinstance(component, CircuitComponent):
+                comps.append(component)
+            else:
+                for id, cmp in self.components.items():
+                    if component in [cmp.id, cmp.name, cmp.composed_name]:
+                        comps.append(cmp)
+                        break
+        i = 0
+        assert len(comps) > 1, "At least two components has to be passed"
```
Comment (MaxJPRey) [inline pyaedt/modeler/PrimitivesNexxim.py:94]:
```suggestion
        assert len(comps) > 1, "At least two components have to be passed."
```

File: pyaedt/modeler/PrimitivesNexxim.py [hunk 2]
```diff
@@ -59,12 +59,88 @@ def __init__(self, modeler):
...
+        comps = []
+        for component in components_to_connect:
+            if isinstance(component, CircuitComponent):
+                comps.append(component)
+            else:
+                for id, cmp in self.components.items():
+                    if component in [cmp.id, cmp.name, cmp.composed_name]:
+                        comps.append(cmp)
+                        break
+        assert len(comps) > 1, "At least two components has to be passed"
```
Comment (MaxJPRey) [inline pyaedt/modeler/PrimitivesNexxim.py:131]:
```suggestion
        assert len(comps) > 1, "At least two components have to be passed."
```

File: pyaedt/modeler/Object3d.py [hunk 3]
```diff
@@ -1883,6 +1883,19 @@ def location(self):
                 _retry_ntimes(30, self.m_Editor.GetComponentPinLocation, self._circuit_comp.composed_name, self.name,
                               False)]
 
+    @property
+    def net(self):
+        """Pin net name."""
```
Comment (MaxJPRey) [inline pyaedt/modeler/Object3d.py:1888]:
```suggestion
        """Get pin net."""
```

File: pyaedt/modeler/Object3d.py [hunk 4]
```diff
@@ -1883,6 +1883,19 @@ def location(self):
...
+        if "PagePort@" in self.name:
+            return self._circuit_comp.name.split("@")[1]
+        for net in self._circuit_comp._circuit_components.nets:
+            conns = self.m_Editor.GetNetConnections(net)
+            for conn in conns:
+                if self.name == conn[-len(self.name):] and (
```
Comment (Alberto-DM) [inline pyaedt/modeler/Object3d.py:1894]:
```suggestion
                if conn.endswith(self.name) and (
```

**Review by Alberto-DM — APPROVED** [review]
small change suggested.
Otherwise LGTM. Approved

**Review by MaxJPRey — APPROVED** [review]

**Comment by MaxJPRey** [comment]
LGTM.

**Pull request merged**